bucket/leaky_bucket_test.go

```diff
@@ -49,12 +51,21 @@ func TestGrabbingADropFromTheBucket(t *testing.T) {
 }
 
 func TestFillingAnAlreadyFullBucket(t *testing.T) {
-	bucketSize := 2
-	bucket := NewLeakyBucket(bucketSize, 0, 0)
+	config := Configuration{Size: 2, Refill: 1, Duration: time.Duration(10) * time.Millisecond}
+	bucket := NewLeakyBucketWithConfiguration(config)
 	bucket.TopUp()
 
 	assert.Equal(t, true, bucket.IsFull())
 	bucket.AddDrops()
 	assert.Equal(t, true, bucket.IsFull())
-	assert.Equal(t, 2, bucket.UnsafeAvailable())
+	assert.Equal(t, 2, bucket.Available())
+}
+
+func BenchmarkBucket(b *testing.B) {
+	config := Configuration{Size: 2, Refill: 1, Duration: time.Duration(1) * time.Millisecond}
```

Comment:
That's a legit reason to do it!

On Fri, Dec 18, 2015 at 10:45 AM, Jakob Külzer <notifications@github.com>
wrote:

> In bucket/leaky_bucket_test.go
> <https://github.com/Shopify/themekit/pull/127#discussion_r48037529>:
>
> >  	bucket.TopUp()
> >
> >  	assert.Equal(t, true, bucket.IsFull())
> >  	bucket.AddDrops()
> >  	assert.Equal(t, true, bucket.IsFull())
> > -	assert.Equal(t, 2, bucket.UnsafeAvailable())
> > +	assert.Equal(t, 2, bucket.Available())
> > +}
> > +
> > +func BenchmarkBucket(b *testing.B) {
> > +	config := Configuration{Size: 2, Refill: 1, Duration: time.Duration(1) * time.Millisecond}
>
> Meh, I thought about benchmarking old vs new, but it's by no means a
> performance critical part. I originally added this so i could run the race
> detector, no to so much to test how fast it is.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/themekit/pull/127/files#r48037529>.
>



-- 
Chris Saunders // Developer @ Shopify